mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -412,19 +412,46 @@ public static String getDefaultValue(org.mule.runtime.extension.api.annotation.p
   }
 
   /**
-   * Tests the given {@code object} to be annotated with {@link org.mule.runtime.extension.api.annotation.param.Optional}.
+     * Returns the default value associated with the given annotation.
+   * <p>
+   * The reason for this method to be instead of simply using
+   * {@link org.mule.sdk.api.annotation.param.Optional#defaultValue()} is a limitation on the Java language to have
+   * an annotation which defaults to a {@code null} value. For that reason, this method tests the default value for equality
+   * against the {@link org.mule.sdk.api.annotation.param.Optional#NULL}. If such test is positive, then
+   * {@code null} is returned.
+   * <p>
+   * If a {@code null} {@code optional} is supplied, then this method returns {@code null}
+   *
+   * @param optional a nullable annotation
+   * @return the default value associated to the annotation or {@code null}
+   */
+  public static String getDefaultValue(org.mule.sdk.api.annotation.param.Optional optional) {
+    if (optional == null) {
+      return null;
+    }
+
+    String defaultValue = optional.defaultValue();
+    return org.mule.sdk.api.annotation.param.Optional.NULL.equals(defaultValue) ? null : defaultValue;
+  }
+
+  /**
+   * Tests the given {@code object} to be annotated with {@link org.mule.runtime.extension.api.annotation.param.Optional} or {@link org.mule.sdk.api.annotation.param.Optional}.
    * <p>
    * If the annotation is present, then a default value is extracted by the rules of
-   * {@link #getDefaultValue(org.mule.runtime.extension.api.annotation.param.Optional)}. Otherwise, {@code null} is returned.
+   * {@link #getDefaultValue(org.mule.runtime.extension.api.annotation.param.Optional)} or {@link #getDefaultValue(org.mule.sdk.api.annotation.param.Optional)}. Otherwise, {@code null} is returned.
    * <p>
    * Notice that a {@code null} return value doesn't necessarily mean that the annotation is not present. It could well be that
    * {@code null} happens to be the default value.
    *
-   * @param object an object potentially annotated with {@link org.mule.runtime.extension.api.annotation.param.Optional}
+   * @param object an object potentially annotated with {@link org.mule.runtime.extension.api.annotation.param.Optional} or {@link org.mule.sdk.api.annotation.param.Optional}
    * @return A default value or {@code null}
    */
   public static Object getDefaultValue(AccessibleObject object) {
-    return getDefaultValue(object.getAnnotation(org.mule.runtime.extension.api.annotation.param.Optional.class));
+    String defaultValue = getDefaultValue(object.getAnnotation(org.mule.sdk.api.annotation.param.Optional.class));
+    if (defaultValue == null) {
```

Comment:
null may either mean that the annotation was no present, or the annotation is present but the parameter has no default value. For the second case, I think that we should not check for the extension-api annotation.

Reply:
You're right. Changed.